noir-projects/aztec-nr/aztec/src/note/lifecycle.nr

```diff
@@ -33,56 +30,44 @@ where
 
     context.push_note_hash(note_hash);
 
-    NoteEmission::new(*note, storage_slot)
+    NoteEmission::new(note, storage_slot, note_hash_counter)
 }
 
-pub fn create_note_hash_from_public<Note, let N: u32>(
```

Comment:
I've nuked the `create_note_hash_from_public ` function which was only used by the stale blacklist token and instead I just push the note hash into the context in the blacklist token directly. This made sense to do in this PR because computing the note hash became straightforward:
<img width="1110" alt="image" src="https://github.com/user-attachments/assets/916b0f4b-a57a-4716-afb5-6e46b80efcf8" />